### PR TITLE
Persist more workflow_job data from webhooks to the runner model

### PR DESCRIPTION
**Pull request by enescakir**
### Add migration file to add workflow_job column to runner

The GithubRunner model retains information about its assigned
workflow_job, which is displayed in the UI. This data also proves useful
when investigating incidents. Previously, we didn't store the job's
status and conclusion, despite needing them on several occasions. We
need to do lots of API calls to fetch this information. I have decided to
persist these as well. However, adding numerous columns about the job to
the runner model didn't seem appropriate. Hence, I decided to
consolidate all workflow-related data into the workflow_job column.

### Persist the workflow_job data from webhooks to the runner model.

Before eliminating individual job data columns, we need to populate our
new, consolidated `workflow_job` data to ensure proper integration
functioning. Given the ephemeral nature of GitHub runners, all runners
will eventually possess `workflow_job`. After this, I will proceed with
deployment.

I have excluded the `.steps` property due to its potential for
containing large amounts of data.
Here is an example of a `workflow_job` payload:

    {
      "workflow_job": {
        "id": 18573479297,
        "run_id": 6828737633,
        "workflow_name": "CI",
        "head_branch": "build-test",
        "run_url": "https://api.github.com/repos/enescakir/ubicloud/actions/runs/6828737633",
        "run_attempt": 1,
        "node_id": "CR_kwDOKTBNac8AAAAEUxDNgQ",
        "head_sha": "2bd13725a66c3f9fc859b5f92e51cc6485a8acb7",
        "url": "https://api.github.com/repos/enescakir/ubicloud/actions/jobs/18573479297",
        "html_url": "https://github.com/enescakir/ubicloud/actions/runs/6828737633/job/18573479297",
        "status": "completed",
        "conclusion": "success",
        "created_at": "2023-11-10T19:11:51Z",
        "started_at": "2023-11-10T19:16:43Z",
        "completed_at": "2023-11-10T19:18:51Z",
        "name": "CI ubicloud",
        "steps": [
          {
            "name": "Set up job",
            "status": "completed",
            "conclusion": "success",
            "number": 1,
            "started_at": "2023-11-10T19:16:42.000Z",
            "completed_at": "2023-11-10T19:16:45.000Z"
          },
          .
          .
          .
          {
            "name": "Stop containers",
            "status": "completed",
            "conclusion": "success",
            "number": 14,
            "started_at": "2023-11-10T19:18:50.000Z",
            "completed_at": "2023-11-10T19:18:50.000Z"
          },
          {
            "name": "Complete job",
            "status": "completed",
            "conclusion": "success",
            "number": 15,
            "started_at": "2023-11-10T19:18:50.000Z",
            "completed_at": "2023-11-10T19:18:50.000Z"
          }
        ],
        "check_run_url": "https://api.github.com/repos/enescakir/ubicloud/check-runs/18573479297",
        "labels": [
          "ubicloud"
        ],
        "runner_id": 957,
        "runner_name": "grb9nkxs40824mag6906fz1wpt",
        "runner_group_id": 1,
        "runner_group_name": "Default"
    }}

### Use workflow_job data from new JSONB column

The previous commit consolidates all data related to workflow jobs into
the `workflow_job` JSONB column. This commit uses data from this column,
rather than from individual columns.

### CODE REVIEW

File: model/github_runner.rb [hunk 1]
```diff
@@ -13,14 +13,14 @@ class GithubRunner < Sequel::Model
...
   end
 
   def job_url
-    "http://github.com/#{repository_name}/actions/runs/#{run_id}/job/#{job_id}"
+    "http://github.com/#{repository_name}/actions/runs/#{workflow_job["run_id"]}/job/#{workflow_job["id"]}"
   end
 
   def runner_url
-    "http://github.com/#{repository_name}/settings/actions/runners/#{runner_id}"
+    "http://github.com/#{repository_name}/settings/actions/runners/#{runner_id}" if runner_id
```
Comment (furkansahin) [inline model/github_runner.rb:24]: is this a related change?

**Review by furkansahin — CHANGES_REQUESTED** [review]
The first commit message can be changed like this
```
- need to do lots of API for to fetch this information.
+ need to do lots of API calls to fetch this information.
```

File: model/github_runner.rb (hunk 1 above)
Comment (enescakir) [reply model/github_runner.rb:24]: I think it's not. Let me move it to another PR

Comment (enescakir) [reply model/github_runner.rb]: moved

**Review by byucesoy — APPROVED** [review]

**Review by furkansahin — APPROVED** [review]

**Review by fdr — APPROVED** [review]
